### ISSUE TITLE
Querydsl JPA predicate on joined attribute generates cross join with spring data jpa

### ISSUE THREAD

**Issue by cpekyaman** [querydsl/querydsl#1833]
I'm trying to use query dsl as predicates for spring data jpa finders. 

As a simple example I tried this:
`QAccountListView accountListView = QAccountListView.accountListView;`
`BooleanExpression queryPredicate = accountListView.owner.customerNumber.eq(customerNumber);`
`List<AccountListView> list = Lists.newArrayList(listViewRepository.findAll(queryPredicate));`

Where AccountListView is an entity with a ManyToOne relation to a Customer (as its owner) and listViewRepository is a spring data jpa repository with querydsl support.

Generated query is something like this:
`select * from account_list_view accountlis0_ cross join customer customer1_ where accountlis0_.owner=customer1_.id and customer1_.customer_number=?`

The end result of query is correct because of the where clause joining owner field and customer but existence of cross join instead of a inner or left joing bothered me a little.

Before diving into spring data side I want to ask if joined fields work correctly this way (I expect at least a left join instead of cross join) ?
I haven't seen any examples which use joined attributes this way without using the JPAQuery (and its methods innerJoin,leftJoin etc.), so I'm not sure if this is how querydsl works or is an issue in spring data jpa querydsl integration

Thanks.

**Comment by Shredder121** [comment]
Could you maybe also post the JPQL instead of the resulting SQL?

**Comment by timowest** [comment]
Querydsl generates only the JPQL part, so we need the JPQL query instead, like @Shredder121 already said.

**Comment by cpekyaman** [comment]
sorry for my late reply; I had to switch to another task for work, so couldn't looked at it.

Also I realized that, I was following one of the relatively older tutorial stuff which uses version  <= 3.7 (which also spring data version we use depends on).
I will retry this again (I hope tomorrow) with the current version (or at least >= 4.0.0). 
In the meantime I also checked a few questions about this "cross join" issue with JPA's own criteria API and I suppose generating a cross join is something expected at least on MSSQL if you don't give explicit joins and use attributes of related entities in where clause.

I'll try both querydsl and JPA criteria API and post the results.

**Comment by cpekyaman** [comment]
It turns out that there may be a problem with mapping or db schema. The query I'm trying was using a view and due a mapping or schema issue it generates a cross join as a query.  We couldn't find the exact cause yet; but the fact is, since the query has a joining where clause the execution is like an inner join.

I've tried different variations of the same search with JPQL, querydsl and criteria. Results are similar if you don't give explicit joins. Since the cross join with a joining where clause = inner join mostly; we defer examining this issue for a later time.

Initially I used view variant with querydsl and another variant with jpql for the comparison by mistake (methods with similar names). This is why I first thought this can be an issue with querydsl.
Sorry for this mistake; there is no issue it seems.
